File: cash.c
#include <stdio.h>
#include <math.h>
#include <cs50.h>
// re-submission following instructions with no
// extraneous information.
// compacted code to minimum
int main()
{
    int cents;    // we will convert dollars into cents

    do
    {
        // get a floating point number
        float dollars = get_float("Change owed: ");
        // round to a positive value
        cents = round(dollars * 100);
    }

    while (cents <= 0);
    // how many quarters can be made
    int quarters = cents / 25;
    // how many dimes can be made
    int dimes = (cents % 25) / 10;
    // how many nickels can be made
    int nickels = ((cents % 25) % 10) / 5;
    // how many pennies can be made
    int pennies = ((cents % 25) % 10) % 5;
    // print sum
    printf("%d\n", quarters + dimes + nickels + pennies);
}

File: cash.py
# cash.py
# Bill Louden
# Program to convert dollars into cents
#

from cs50 import get_float


def main():
    money = get_float("Change owed: ")

    while (money < 0):
        money = get_float("Change: ")
        break

    cents = round(money * 100)
#   print (cents)
    quarters = int(cents / 25)
#   print(quarters)
    dimes = int(quarters % 10)
#   print(dimes)
    nickles = int(dimes % 5)
#   print(nickles)
    pennies = int(cents % 25 % 10 % 5)
#   print(pennies)

    print(quarters+dimes+nickles+pennies)


if __name__ == "__main__":
    main()